File: UCF HSPT Documents/2008/Solutions/chalice.cpp

#include <stdio.h>

int main(void)
{
   FILE *fp;
   int  numTrials;
   int  trial;
   int  witchWeight;
   int  numGeese;
   int  i;
   int  gooseWeight;
   int  geeseWeight;

   // Open the input file
   fp = fopen("chalice.in", "r");

   // Read the number of trials
   fscanf(fp, "%d\n", &numTrials);

   // Conduct the trials
   for (trial = 0; trial < numTrials; trial++)
   {
      // Read the weight of the accused
      fscanf(fp, "%d\n", &witchWeight);

      // Read the number of geese
      fscanf(fp, "%d\n", &numGeese);

      // Read and accumulate the weights of the geese
      geeseWeight = 0;
      for (i = 0; i < numGeese; i++)
      {
         // Read the weight
         fscanf(fp, "%d", &gooseWeight);

         // Add it to the total
         geeseWeight += gooseWeight;
      }

      // Compare the two weights and print the output
      if (witchWeight <= geeseWeight)
         printf("Trial #%d: SHE'S A WITCH! BURN HER!\n", trial+1);
      else
         printf("Trial #%d: She's not a witch. BURN HER ANYWAY!\n", trial+1);
   }

   // Close the input file
   fclose(fp);
}

File: UCF HSPT Team/src/y2008/chalice.java
//Montgomery Anaconda and the Sacred Chalice judge solution
//Shehan Sirigampola

import java.io.*;
import java.util.*;

public class chalice {
	public static void main(String[] args) throws IOException, FileNotFoundException {
		FileReader file = new FileReader("chalice.in");
		BufferedReader reader = new BufferedReader(file);
		
		int cases = Integer.parseInt(reader.readLine());				//Input the number of cases and store it in cases
		for(int i = 0; i < cases; i++) {								//Loop through all the cases
			int weight = Integer.parseInt(reader.readLine());			//Input the weight of the accused and store it in weight
			
			int geese = Integer.parseInt(reader.readLine());			//Input the number of geese and store it in geese
			int sum = 0;												//Stores the total weight of the flock of geese
			StringTokenizer tokens = new StringTokenizer(reader.readLine());		//Read in the line with the weight of the geese and tokenize it
			for(int j = 0; j < geese; j++) {							//Loop through all the geese
				sum += Integer.parseInt(tokens.nextToken());			//Add the weight of one goose to the weight of the flock
			}
			
			System.out.print("Trial #" + (i+1) + ": ");					//Output trial number
			
			//Based on weight comparison output whether she is or isn't a witch
			if(weight <= sum)
				System.out.println("SHE'S A WITCH! BURN HER!");
			else
				System.out.println("She's not a witch. BURN HER ANYWAY!");
		}
	}
}
		
